src/host/util/PersistentStorage.h

```diff
@@ -67,7 +67,7 @@ class PersistentStorage {
 
     struct IMUCalibration {
       uint8_t mountingPosition;
-      uint8_t calibrationData[22]; // NUM_BNO055_OFFSET_REGISTERS
+      int calibrationData[22]; // NUM_BNO055_OFFSET_REGISTERS
```

Comment:
why int? it's an `int16_t`, no?

That is a huge bug - and it was all me! Definitely explains some of the weird results we were seeing. Thanks a lot for finding that - and sorry for screwing up your code ;)

Reply:
Actually, there is no bug.

We are using the function `getSensorOffsets(uint8_t* calibData)` which reads the register as an array and writes them back as an array so the changes Adafruit did will not impact us. And we are reading the correct size: 22 registers of 8bits each.

So I do not think this PR will improve anything unfortunately.

We can merge the Adafruit update to stay current but that will not change anything for us.